app/controllers/posts_controller.rb

```diff
@@ -65,34 +67,36 @@ def show
   end
 
   def update
-    post = current_organization.posts.find params[:id]
+    post = @organization.posts.find params[:id]
     authorize post
     instance_variable_set("@#{resource}", post)
     if post.update_attributes(post_params)
-      redirect_to post
+      redirect_to polymorphic_url([@organization, post])
     else
       render action: :edit, status: :unprocessable_entity
     end
   end
 
   def destroy
-    post = current_organization.posts.find params[:id]
+    post = @organization.posts.find params[:id]
     authorize post
-    redirect_to send("#{resources}_path") if post.update!(active: false)
+    redirect_to polymorphic_url([@organization, resources]) if post.update!(active: false)
   end
 
   private
 
+  # TODO: Investigate why we need this
   def resource
     controller_name.singularize
   end
 
+  # TODO: Investigate why we need this
   def resources
     controller_name
   end
 
   def set_user_id(p)
-    if current_user.manages?(current_organization)
+    if current_user.manages?(@organization)
```

Comment:
we could avoid lots of these ivar changes by implemanting the getter method in the application controller as:

```ruby
def current_organization
  @organization
end
```

This would reduce the number of changes considerable and increase the readability of this PR.